newIDE/app/src/SceneEditor/index.js

```diff
@@ -386,7 +402,31 @@ export default class SceneEditor extends Component {
     done(true);
   };
 
-  _onRenameObject = (objectWithContext, newName, done) => {
+  //use this function to check if the new name can be used
+  // on apply new name event
+  _canObjectUseNewName = (objectWithContext, newName) => {
+    const { project, layout } = this.props;
+    const { object } = objectWithContext;
+
+    if (object.getName() === newName) {
+      return true;
+    } else if (
+      layout.hasObjectNamed(newName) ||
+      project.hasObjectNamed(newName)
+    ) {
+      showWarningBox('Another object with this name already exists.');
+      return false;
+    }
+    return true;
+  };
+
+  _onRenameEditedObject = newName => {
+    if (newName !== null) {
```

Comment:
You don't need the comparison with `!== null`, as newName is always a string

Reply:
done:)

Reply:
Actually- if you dont pass an optional callback function (in my case I didnt need one), this line caused gdevelop to crash

Reply:
Are we speaking of the same line? This is the line for the name of the object. You don't need the null here, as the object name should never be null.

Reply:
aah sorry, I was looking at another thing. You are right - the null check there is absolutely not required

Reply:
fixed now